### PR TITLE
Add IRSA to Upbound Provider

### PR DESCRIPTION
**Pull request by blakeromano**
### What does this PR do?

This translates 

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)

- [x] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR

- [x] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature

- [ ] Yes, I have linked to an issue or feature request (applicable to PRs that solves a bug or a feature request)

**Note**:
 - Not all the PRs require examples and docs
 - We prefer small, well tested pull requests. Please ensure your pull requests are self-contained, and commits are squashed

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: compositions/upbound-aws-provider/irsa/irsa-exact.yaml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                  ]
+                }
+    - name: policy-attachment-1
+      base:
+        apiVersion: iam.aws.upbound.io/v1beta1
+        kind: RolePolicyAttachment
+        spec:
+          forProvider:
+            policyArn: ""
+            roleNameSelector:
```
Comment (nabuskey) [inline compositions/upbound-aws-provider/irsa/irsa-exact.yaml:79]:
```suggestion
            roleSelector:
```

File: compositions/upbound-aws-provider/irsa/irsa-exact.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+          fromFieldPath: spec.policyArns[0]
+          toFieldPath: spec.forProvider.policyArn
+    - name: policy-attachment-2
+      base:
+        apiVersion: iam.aws.upbound.io/v1beta1
+        kind: RolePolicyAttachment
+        spec:
+          forProvider:
+            policyArn: ""
+            roleNameSelector:
```
Comment (nabuskey) [inline compositions/upbound-aws-provider/irsa/irsa-exact.yaml:94]:
```suggestion
            roleSelector:
```

**Review by nabuskey — CHANGES_REQUESTED** [review]
Thanks for this. I've tested it and ran into some issues. Selector field names have changed so we do need to update it, otherwise nothing gets selected.

File: compositions/upbound-aws-provider/irsa/irsa-exact.yaml (hunk 1 above)
Comment (blakeromano) [reply compositions/upbound-aws-provider/irsa/irsa-exact.yaml]: Ahh good catch I fixed!

**Review by nabuskey — APPROVED** [review]
Looks good. I will need to add another claim examples and specify compositions but we can do that later.

**Pull request merged**